### PR TITLE
tests: Add --debug option

### PR DESCRIPTION
**Pull request by mlouielu**
This will let each test case run with "dbg",
so the test case will wait for gdb connect.

### CODE REVIEW

**Comment by jserv** [comment]
Please resolve the conflicts after commit 71a165cb2dcb5b0223c31716184287f08fc6a7d2 was introduced.

File: tests/runner.py [hunk 1]
```diff
@@ -37,10 +37,10 @@ def print_header(testname, arch):
     print("time        :  %s\n" % strftime("%c"))
 
 
-def run_single_test(testname, verbose, platform='stm32p103'):
+def run_single_test(testname, verbose, platform='stm32p103', debug=False):
     # platform = os.getenv('PLATFORM', 'qemu')
     cmd = ["make", "TEST=%s" % testname, "TARGET=%s" % platform,
-           "--file", "tests/Makefile", "clean_test", "all", "run"]
+           "--file", "tests/Makefile", "clean_test", "all", "dbg" if debug else "run"]
```
Comment (jserv) [inline tests/runner.py]: How about changing from 'dbg' to 'debug' for better expression?

Comment (mlouielu) [reply tests/runner.py]: I should not do this, but using argparse `args` to set up.

**Comment by jserv** [comment]
Rebase is required before the merge.

File: tests/runner.py (hunk 1 above)
Comment (mlouielu) [reply tests/runner.py]: I'll fix this in different PR.

**Comment by mlouielu** [comment]
Done.

**Pull request merged**